sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java

```diff
@@ -63,30 +65,39 @@ public static synchronized void getDbToken(
             TokenExchangeServiceCallback serviceCallback) {
         AppCenterLog.debug(LOG_TAG, "Getting a resource token from App Center...");
         String url = apiUrl + GET_TOKEN_PATH_FORMAT;
-
-        ServiceCall tokenResponse =
-            httpClient.callAsync(
-                url,
-                METHOD_POST,
-                new HashMap<String, String>() { { put(APP_SECRET_HEADER, appSecret); } },
-                    new HttpClient.CallTemplate() {
-
-                        @Override
-                        public String buildRequestBody() {
-                            return buildAppCenterGetDbTokenBodyPayload(partition);
-                        }
-
-                        @Override
-                        public void onBeforeCalling(URL url, Map<String, String> headers) { }
-                    },
-                    serviceCallback);
+        TokenResult tokenResult = TokenManager.getInstance().getCachedToken(partition);
+        if (tokenResult != null) {
+            serviceCallback.callCosmosDb(tokenResult);
+        } else {
+            ServiceCall tokenResponse =
+                    httpClient.callAsync(
+                            url,
+                            METHOD_POST,
+                            new HashMap<String, String>() { { put(APP_SECRET_HEADER, appSecret); } },
+                            new HttpClient.CallTemplate() {
+
+                                @Override
+                                public String buildRequestBody() {
+                                    return buildAppCenterGetDbTokenBodyPayload(partition);
+                                }
+
+                                @Override
+                                public void onBeforeCalling(URL url, Map<String, String> headers) { }
+                            },
+                            serviceCallback);
+        }
     }
 
     public abstract static class TokenExchangeServiceCallback implements ServiceCallback {
         @Override
         public void onCallSucceeded(String payload, Map<String, String> headers) {
             final TokenResult tokenResult = parseTokenResult(payload);
-            callCosmosDb(tokenResult);
+            if (!tokenResult.status().equals(Constants.SUCCEED)){
```

Comment:
If you want to make the status case-insensitive, you can use all lower case for the constant and use `equalsIgnoreCase`. FYI.